software-requirement-specification.md

```diff
@@ -99,29 +106,37 @@ This section describes specific features of the software project. Specify the re
 
 ### 3.2 Non-Functional Requirements
 
-Non-functional requirements may exist for the following attributes. Often these requirements must be achieved at a system-wide level rather than at a unit level. State the requirements in the following sections in measurable terms (e.g., 95% of transactions shall be processed in less than a second, system downtime may not exceed 1 minute per day, > 30-day MTBF value, etc).
-
-This section describes non-functional features of the software project. Specify the requirements as user stories.
+This section outlines the non-functional requirements crucial for the U↑ internet-based phone system, focusing on performance, reliability, availability, security, and interfaces.
 
 #### 3.2.1 Performance
 
-- The product shall take initial load time depending on internet connection strength which also depends on the media from which the product is run.
+- **System Latency:** Call rerouting should happen within a 5-second window.
+- **Concurrency:** The system must support at least 500 concurrent calls without degradation in performance.
+- **Scalability:** The system should accommodate a 20% increase in user devices annually without requiring a system overhaul.
 
 #### 3.2.2 Reliability
 
-- The system shall provide RAID V Disk Stripping on all database storage disks.
+- **Data Integrity:** The system must achieve a data accuracy rate of 99.999%.
+- **Backup:** Weekly automated backups are mandatory, with a successful restore rate of at least 99%.
 
 #### 3.2.3 Availability
 
-- The system shall provide a contractual agreement with an internet service provider for T3 access with 99.9999% availability.
+- **Uptime:** The system should guarantee 99.9% uptime, aside from planned maintenance.
+- **Failover Capabilities:** Advanced call routing algorithms and fallback mechanisms must be in place to manage unforeseen call scenarios.
+- **Maintenance Window:** Planned maintenance must be confined to off-peak hours, occurring no more than once a month.
 
 #### 3.2.4 Security
 
-- The system shall use secure sockets in all transactions that include any confidential customer information.
-- The system shall automatically log out all customers after a period of inactivity.
+- **Data Encryption:** All calls must be end-to-end encrypted.
+- **Authentication:** Multi-factor authentication must be implemented for both Administrators and regular users.
+- **Session Timeout:** Administrator sessions in the User Interface should automatically timeout after 30 minutes of inactivity.
 
 #### 3.2.5 Interfaces
 
+- **User Interface:** The UI must be intuitive and require no more than one hour of training for efficient utilization.
+- **Onboarding Time:** New users should be able to complete the onboarding process within 20 minutes, on average.
+- **Accessibility:** The system should be usable across desktop computers, smartphones, tablets, and other devices.
+
```

Comment:
This part was 'technically' cody's but he's pr is a bit different so we'll see

Reply:
Was this not supposed to be under section 3.2?